packages/diracx-web-components/components/shared/FilterToolbar.tsx

```diff
@@ -23,16 +24,26 @@ interface FilterToolbarProps {
   appliedFilters: InternalFilter[];
   /** The function to apply the filters */
   handleApplyFilters: () => void;
+  /** The function to remove all filters */
+  handleClearFilters: () => void;
 }
 
 /**
  * Filter toolbar component
  *
  * @returns a FilterToolbar component
  */
-export function FilterToolbar(props: FilterToolbarProps) {
-  const { columns, filters, setFilters, appliedFilters, handleApplyFilters } =
-    props;
+export function FilterToolbar<T extends Record<string, unknown>>(
+  props: FilterToolbarProps<T>,
+) {
+  const {
+    columns,
+    filters,
+    setFilters,
+    appliedFilters,
+    handleApplyFilters,
+    handleClearFilters,
+  } = props;
```

Comment:
how about something like this?

```suggestion
export function FilterToolbar<T extends Record<string, unknown>>(
  {
    columns,
    filters,
    setFilters,
    appliedFilters,
    handleApplyFilters,
    handleClearFilters,
  }: FilterToolbarProps<T>,
) {
```

Reply:
From what I understand (again, don't hesitate to correct me if you think I am wrong), this also comes down to personal preference most of the time. Now in our case, I think the structured props has a few advantage over the "destructured" one:
- in your suggestion, you did not add the types of the parameters, and they would make the signature less readable given the number of parameters.
- the props and the comments associated are used by `Storybook` to provide an interactive documentation about the components for the communities that will have to develop their extensions (e.g. LHCb), I am not sure (I have not checked though) `Storybook` would be able to read the parameters from there.

Again, no matter our final choice, may be it would deserve a consistency check? (I think there is a `destructuring` check for that)

Reply:
> you did not add the types of the parameters

I'd thought that i'd specified the types (line 44) `}: FilterToolbarProps<T>`


No biggie if you prefer the current form - I'd thought it might be a nicer way the component props